arangod/Aql/QuerySnippet.cpp

```diff
@@ -403,11 +403,14 @@ ResultT<std::unordered_map<ExecutionNode*, std::set<ShardID>>> QuerySnippet::pre
           // OneShard case) because local graphs (on DBServers) only ever occur
           // in either OneShard or SatelliteGraphs.
           TRI_ASSERT(found->second == server || localGraphNode->isUsedAsSatellite() ||
-                     aqlCollection->isSatellite());
+                     aqlCollection->isSatellite() || localGraphNode->isDisjoint());
           // provide a correct translation from collection to shard
           // to be used in toVelocyPack methods of classes derived
           // from GraphNode
-          localGraphNode->addCollectionToShard(aqlCollection->name(), shard);
+          if (found->second == server || !localGraphNode->isDisjoint()) {
+            // TODO: Optimize this, we're looking into to many collections here
```

Comment:
*when* do we optimize this? Do we have at least a Kanban card for it?

Reply:
Yes we have